### ISSUE TITLE
Customsanitizer not called

### ISSUE THREAD

**Issue by orlavy-bigabid** [express-validator/express-validator#592]
As part of my route validation/sanitize chain I am using the following line : 
<code> 
body('source.id').exists().isNumeric().toInt().customSanitizer(id => id === newSourceId ? genId() : id)
 </code>

That line used to work and recently some weird bugs started to appear,
after checking I found out that the 'customSanitizer' lambda is never called

**Comment by gustavohenke** [comment]
Please explain this further. This is too brief and I can't do a thing about it.

Try answering these questions:
- What version are you using?
- Have you updated recently? From which version?
- What is your request body like?
- Did you change your code recently?

**Comment by cyrillef** [comment]
Same problem here with a json payload (Content-Type: application/json). Using Node.js v9.5, express v4.16.3, express-validator v5.2.0

```
{
  field1: 10,
  field2: "a string",
  field3: [ 1, 2, "3" ],
  field4: {
    field4_1: 20,
    field4_2: "another string"
  }
}
```
Now the check & sanitize. Everything in comment is never called for a json payload. Would be called for a form-urlencoded body.

```
[checkSchema({
  field1: { in: ['body'],
    exists: true,
    isInt: true,
    errorMessage: 'field1 is missing or not an int'
  },
  'field2': { in: ['body'],
    exists: true,
    isString: true,
    isEmpty: false,
    errorMessage: 'field2 is missing or empty'
  },
  field2: { in: ['body'],
    optional: true,
    isArray: true,
    custom: {
      options: (value, { req, location, path }) => {
        let result = value.every(function (val) {
          return (validator.isInt(val.toString()));
        });
        return (result);
      }
    },
    // customSanitizer: (value, { req, location, path }) => {
    // 	let result = value.every(function (val) {
    // 		return (validator.toInt(val));
    // 	});
    // 	return (result);
    // },
    errorMessage: 'field2 is present but malformed'
  },
  field4: { in: ['body'],
    exists: true,
    errorMessage: 'field4 is missing'
    },
  'field4.field4_1': { in: ['body'],
    exists: true,
    isInt: true,
    errorMessage: 'field4_1 is missing or not an int'
  },
  'field4.field4_2': { in: ['body'],
    optional: true,
    isString: true,
    errorMessage: 'field4_2 is not a string'
  }
}),
// sanitizeBody('field1').customSanitizer ((value, { req, location, path }) => {
//	return (value);
// }),
// sanitizeBody('field4.field4_1').customSanitizer ((value, { req, location, path }) => {
//	return (value);
// })
]
```

The workaround I am using for now is:

```
sanitize('any_parameter') // any parameter I got in my request
  .customSanitizer ((value, { req, location, path }) => {
    let _body = matchedData(req, { locations: ['body'], path: 'field2' });
    _body.field2.map((val, index) => {
      if ( typeof val !== 'string' )
        return (val) ;
      _body.field2 [index] =validator.toInt(val); // modifies body correctly.
    });
    return (value);
  });

**Comment by jose-nunez** [comment]
Hi there. I'm having the same problem here. 
Custom checkers work ok for me:
```
const { check } = require('express-validator/check')
check('myParam').optional().custom(customChecker)
```
Custom sanitizer is not being called here:
```
const { check } = require('express-validator/check')
check('myParam').optional().custom(customChecker).customSanitizer(customSanitizer)
```
Here neither:
```
const { check } = require('express-validator/check')
const { sanitize } = require('express-validator/filter')
check('myParam').optional().custom(customChecker)
sanitize('myParam').customSanitizer(customSanitizer) // This doesn't work
```

I'm running:
Node v8.7.0
express v4.16.3
express-validator v5.2.0

**Comment by orlavy-bigabid** [comment]
great find @alzeih ! 

If it is indeed the case, I think that we need also the capability to be able to validate any data type.

**Comment by alzeih** [comment]
It looks like there’s some progress towards this idea at #641 ?

**Comment by gustavohenke** [comment]
Hello folks! I'm sorry for this less than great response time from me.
A fix has been published in [`v6.0.0`](https://github.com/express-validator/express-validator/releases/tag/v6.0.0) 🚢, please try it out!
